packages/electron-commons/test/initializers/electron-node.spec.ts

```diff
@@ -28,16 +29,17 @@ const setupRunningEnv = async ({
             config: [testFeature.use({ errorType: { type: errorMode, handleUncaught } })],
             features: Array.from(features.entries()),
         },
-        processOptions: { cwd: process.cwd(), stdio: ['ignore', 'ignore', 'ignore', 'ipc'] },
+        processOptions: {
+            cwd: process.cwd(),
+            stdio: pipeOutput === true ? ['pipe', 'pipe', 'pipe', 'ipc'] : ['ignore', 'ignore', 'ignore', 'ipc'],
+        },
         environmentStartupOptions: {},
     });
 
     await environmentIsReady;
 
-    const disconnectPromise = new Promise<boolean>((res) => {
-        onDisconnect(() => {
-            res(true);
-        });
+    const disconnectPromise = new Promise<DisconnectDetails>((res) => {
```

Comment:
the mixture of the api accepting a callback and us wanting to use promises in async flows is really shown here.

there are several ways we could alleviate this:
- expose the child process to outside to listen to exit on its own terms.
- use node's event emitter for these events with tests able to call `{once} from 'events'`.
- error handling, should be done via IPC and exposed as on-going error events every time an error occurs inside the child process

Reply:
This is just about how we are waiting process exit to test it. I'd not put too much attention to this.
also, please note that I haven't introduced any new approach here. this particular thing was like this for a long time. In this PR I've added just process exit details that has no relation to promises here.
however, changed this to deferred for more readability